src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Read.cs

```diff
@@ -538,21 +536,24 @@ private void ReadPosixAndGnuSharedAttributes(ReadOnlySpan<byte> buffer)
         // Throws if any conversion fails.
         private void ReadGnuAttributes(ReadOnlySpan<byte> buffer)
         {
-            // Convert byte arrays
-            ReadOnlySpan<byte> aTimeBuffer = buffer.Slice(FieldLocations.ATime, FieldLengths.ATime);
-            if (!aTimeBuffer.SequenceEqual(ArrayOf12NullBytes)) // null values are ignored
-            {
-                long aTime = TarHelpers.ParseNumeric<long>(aTimeBuffer);
-                _aTime = TarHelpers.GetDateTimeOffsetFromSecondsSinceEpoch(aTime);
-            }
-            ReadOnlySpan<byte> cTimeBuffer = buffer.Slice(FieldLocations.CTime, FieldLengths.CTime);
-            if (!cTimeBuffer.SequenceEqual(ArrayOf12NullBytes)) // An all nulls buffer is interpreted as MinValue
+            _aTime = ParseAsTimestamp(buffer.Slice(FieldLocations.ATime, FieldLengths.ATime));
+
+            _cTime = ParseAsTimestamp(buffer.Slice(FieldLocations.CTime, FieldLengths.CTime));
+
+            // TODO: Read the bytes of the currently unsupported GNU fields, in case user wants to write this entry into another GNU archive, they need to be preserved. https://github.com/dotnet/runtime/issues/68230
+        }
+
+        private static DateTimeOffset ParseAsTimestamp(ReadOnlySpan<byte> buffer)
+        {
+            // When all bytes are zero, the timestamp is not initialized, and we map it to default.
```

Comment:
Technically is an all-zero timestamp a valid timestamp?

Reply:
No, a valid timestamp is never serialized as all zero bytes.